core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -118,17 +118,13 @@ public static void schedule(String threadName) {
   }
 
   /**
-   * Waits until the given thread can be executed.
+   * Waits for up to 10 seconds until the given thread can be executed.
    *
    * @param name a name of the thread to wait for
    * @throws InterruptedException if the waiting thread is interrupted
    */
   public static void await(String name) throws InterruptedException {
-    try (LockResource r = new LockResource(sLock)) {
-      while (!sTimers.containsKey(name)) {
-        sCondition.await();
-      }
-    }
+    await(name, 10, TimeUnit.SECONDS);
```

Comment:
Right, I was suggesting changing the name to avoid the anti-symmetry.

I'm mostly looking at this from the perspective of building on Jenkins. If a test uses await() without a timeout, it will hang for two hours and no further tests will be run. It would be better if it would time out and allow other tests to run. One thing we could do here is increase the default timeout to a large value like 5 minutes so that it's clear that the test failed due to hanging as opposed to timing issues.

The problem with method-level timeouts is that they don't make it clear where the hang is happening and it isn't clear why they are being used (is it there in case of hangs, or is this a performance test, etc.). I'd rather not add method-level timeouts to every unit test.

Anyway, I don't think this is a big deal so I'll switch to hanging the build forever when the heartbeat hangs.